hhub/views.py

```diff
@@ -108,3 +121,24 @@ def search_entries(request):
 
 def entry_platform(request, platform):
     return
+
+#########
+# UTILS #
+#########
+def sort_and_order(entries, order_by, sort_by):
+    try:
+        # order_by a specific field from schema
+        if order_by != 1 and order_by != "":
+            # specify an order --> asc is by default, so we're not dealing with it
+            if sort_by != 1 and sort_by != "" and sort_by.lower() != "asc":
+                return entries.order_by("-" + order_by.lower())  # order_by desc order
+            else:
+                return entries.order_by(order_by.lower())        # by default: sort by asc order
+        else:
+            # TODO: should we notify the user about a malformed query in case of empty order_by param? e.g. /api/all?order_by=&page=2
+            return entries
+
+    except FieldError as e:
```

Comment:
Why and when this happens? Why is the user allowed to query everything on the db?

Reply:
when user writes gibberish in group_by (non-existent field); if you try it out, you will find that Django will give an error.

Reply:
After testing once more, I found that I was missing something else.

FIY: if we call http://127.0.0.1:8000/api/all?order_by=djiwawdij&page=1
{"error": "Cannot resolve keyword 'djiwawdij' into field. Choices are: developer, platform, slug, title, typetag"}

Reply:
Django gives error because the column doesn't exist. The user shouldn't be allowed to choose from everything, but just from a limited subset of columns..

Reply:
I agree with you; what's the best solution to achieve this in your opinion?
And what columns should be available?

Reply:
Maybe an "allowed categories" (developer, platform, slug, title, typetag) and then run the order_by only if the request column is among in the accepted?